E2E failure on common TestRunExtraRepository test

### What happened?

The common e2e test TestRunExtraRepository has been failing for some time.

Some jakarta missing class can be seen in the logs.

### Steps to reproduce

Look into a github action logs ([common](https://github.com/apache/camel-k/actions/workflows/common.yml?query=branch%3Amain)) .

### Relevant log output

```shell
2023-09-28T15:53:29Z	DEBUG	camel-k	Finding available platforms
  2023-09-28T15:53:29Z	DEBUG	camel-k	Found active local integration platform test-4cde0787-4629-4829-9e96-005fb2980cf7
  Integration "java" created
      maven_repository_test.go:47: 
          Timed out after 60.001s.
          Expected
.....
3.0.3.jar:dependencies/lib/main/org.eclipse.microprofile.context-propagation.microprofile-context-propagation-api-1.3.jar:dependencies/lib/main/org.jboss.logging.commons-logging-jboss-logging-1.0.0.Final.jar:dependencies/lib/main/org.jboss.logging.jboss-logging-annotations-2.2.1.Final.jar:dependencies/lib/main/org.jboss.slf4j.slf4j-jboss-logmanager-2.0.0.Final.jar:dependencies/lib/main/org
  .jboss.threads.jboss-threads-3.5.0.Final.jar:dependencies/lib/main/org.jooq.joor-0.9.14.jar:dependencies/lib/main/org.slf4j.slf4j-api-2.0.6.jar:dependencies/lib/main/org.snakeyaml.snakeyaml-engine-2.6.jar:dependencies/lib/main/org.yaml.snakeyaml-2.0.jar:dependencies/quarkus-app-dependencies.txt:dependencies/quarkus-run.jar:dependencies/quarkus/generated-bytecode.jar:dependencies/quarkus/quarkus-application.dat:dependencies/quarkus/transformed-bytecode.jar io.quarkus.bootstrap.runner.QuarkusEntryPoint
              Exception in thread "main" java.lang.reflect.InvocationTargetException
              	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
              	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
              	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
              	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
              	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
              	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
              Caused by: java.lang.NoClassDefFoundError: jakarta/annotation/Priority
              	at io.smallrye.config.SmallRyeConfigBuilder.getPriority(SmallRyeConfigBuilder.java:524)
              	at io.smallrye.config.SmallRyeConfigBuilder.addConverter(SmallRyeConfigBuilder.java:510)
....
              	... 6 more
              Caused by: java.lang.ClassNotFoundException: jakarta.annotation.Priority
              	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
              	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
              	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
              	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:115)
              	at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:65)
```


### Camel K version

main branch

--- a/e2e/common/misc/maven_repository_test.go
+++ b/e2e/common/misc/maven_repository_test.go
@@ -38,7 +38,7 @@ func TestRunExtraRepository(t *testing.T) {
 	name := "java"
 	Expect(KamelRunWithID(operatorID, ns, "files/Java.java",
 		"--maven-repository", "https://maven.repository.redhat.com/ga@id=redhat",
-		"--dependency", "mvn:jakarta.annotation:jakarta.annotation-api:1.3.5.redhat-00002",
+		"--dependency", "mvn:org.jolokia:jolokia-core:1.7.1.redhat-00001",
 		"--name", name,
 	).Execute()).To(Succeed())
 

